### DISCUSSION TITLE
G29 command not working on Ender 3 V2

### DISCUSSION THREAD

**Discussion by AradiFerenc** [mriscoc/Ender3V2S1#997] [Q&A] [answered]
Hello Everyone!

I just bought a CR Touch and when it arrived I wanted to use this firmware as well, so I followed the wiki to install and it booted alright.
Now when I open pronterface and connect my printer, then sending the G29 command for bed leveling does not seem to do anything (I did home my machine beforehand). I used jyers firmware beforehand to level my bed manually with positioning assistance. When I go into advanced settings/mesh leveling/auto build mesh it goes around and does a mesh but does not seem to use that mesh when printig even with the M420 S1 Z2 line in my gcode. I tried flashing the jyers firmware with auto bed leveling for the CR Touch and then G29 worked perfectly fine. I also tried resetting all settings there, then installing the stock firmware (G29 also worked here), also resetting settings, then installing this firmware and the problem still persists.

I could not find this bug in the issues tab, so I'm not sure if I'm the only person having this issue, or I'm just doing something wrong, because I am new to the world of 3D printing.

Does anyone know what can be the problem or how can I solve this?

I tried:
* Resetting all settings.
* Installing stock firmware, resetting all settings then reinstalling this firmware.
* Installing jyers firmware, resetting all settings then reinstalling this firmware.
* G29 works when stock firmware is installed
* G29 works when jyers firmware is installed

Printer:
* Stock ender 3 V2 except for the freshly installed CR Touch sensor
* Main board version 4.2.2
* Board processor type RET6

Firmware I tried installing:
* Ender3V2-422-BLTUBL-MPC-20230522B.bin

**Comment by Crazy-Charles** [comment, marked as answer]
When using UBL you need to use "G29 P1" to build the mesh  https://marlinfw.org/docs/gcode/G029-ubl.html

**Reply by AradiFerenc** [reply]
Thank you, this fixed my problem! :)
